src/v/test_utils/test.h

```diff
@@ -145,6 +145,16 @@ class seastar_test : public ::testing::Test {
     GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_FATAL_FAILURE_CORO_)
 #define GTEST_ASSERT_EQ_CORO(val1, val2)                                       \
     ASSERT_PRED_FORMAT2_CORO(::testing::internal::EqHelper::Compare, val1, val2)
+#define GTEST_ASSERT_GT_CORO(val1, val2)                                       \
+    ASSERT_PRED_FORMAT2_CORO(::testing::internal::CmpHelperGT, val1, val2)
+#define GTEST_ASSERT_GE_CORO(val1, val2)                                       \
+    ASSERT_PRED_FORMAT2_CORO(::testing::internal::CmpHelperGE, val1, val2)
+#define GTEST_ASSERT_LT_CORO(val1, val2)                                       \
+    ASSERT_PRED_FORMAT2_CORO(::testing::internal::CmpHelperLT, val1, val2)
+#define GTEST_ASSERT_LE_CORO(val1, val2)                                       \
+    ASSERT_PRED_FORMAT2_CORO(::testing::internal::CmpHelperLE, val1, val2)
+#define GTEST_ASSERT_NE_CORO(val1, val2)                                       \
+    ASSERT_PRED_FORMAT2_CORO(::testing::internal::CmpHelperNE, val1, val2)
```

Comment:
nice.

cc @rockwotj

Reply:
Cool, maybe I need to start using coroutines in tests 😄 

FWIW this double declaring of `GTEST_ASSERT_GT_CORO` then `ASSERT_GT_CORO` probably isn't needed for us. GTest only has this because it supports disabling the non prefixed macros if there are conflicting names, so we can probably save some boilerplate for adding new ones of these.

Reply:
Ah TIL, thanks for the context! Maybe I'll do this as follow-up.

Yeah TBH I added these and ended up not using them. A lot of our unit test fixtures/utilities have `.get()` sprinkled in (looking at you, `redpanda_thread_fixture`), it mixing them together pretty challenging.

Reply:
> Maybe I'll do this as follow-up.

Yeah I had no intentions of asking you to do anything here, just wanted to let you know